In [2]:
import pandas as pd
import numpy as np

In [3]:
df=pd.read_excel("C:\\Users\\dzjc9\\Desktop\\Forex_Analysis\\bog.xlsx")

In [4]:
df


Date   Currency Currency Pair      Buying     Selling    Mid Rate
0     06 Jan 2022  US Dollar        USDGHS      6.0034      6.0094      6.0064
1     05 Jan 2022  US Dollar        USDGHS      6.0029      6.0089      6.0059
2     04 Jan 2022  US Dollar        USDGHS      6.0031      6.0091      6.0061
3     31 Dec 2021  US Dollar        USDGHS      6.0031      6.0091      6.0061
4     30 Dec 2021  US Dollar        USDGHS      5.9931      5.9991      5.9961
...           ...        ...           ...         ...         ...         ...
6474  08 Jan 1996  US Dollar        USDGHS  1,440.9100  1,467.7300  1,454.3200
6475  05 Jan 1996  US Dollar        USDGHS  1,442.2700  1,469.0900  1,455.6800
6476  04 Jan 1996  US Dollar        USDGHS  1,438.6400  1,465.9100  1,452.2750
6477  03 Jan 1996  US Dollar        USDGHS  1,437.2700  1,464.3600  1,450.8150
6478  02 Jan 1996  US Dollar        USDGHS  1,434.0900  1,461.6400  1,447.8650

[6479 rows x 6 columns]

The Data df has 6 columns, the Date, the currency, currency pair,Buying, selling and Mid Rate.


Date shows the occassion or day of every transaction

Currency displays the currency which is being matched up with the Ghana Cedi

Currency pair, when an order is placed for a currency pair, the first listed currency or base currency is bought while the second listed currency in a currency pair or quote currency is sold.

The buying rate is the rate at which money dealers will buy foreign currency

The selling rate is the rate at which they will sell that currency.

The mid and mid-market rate, is the exchange rate exactly halfway between a currency's bid and ask rates.


In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6479 entries, 0 to 6478
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Date           6479 non-null   object
 1   Currency       6479 non-null   object
 2   Currency Pair  6479 non-null   object
 3   Buying         6479 non-null   object
 4   Selling        6479 non-null   object
 5   Mid Rate       6479 non-null   object
dtypes: object(6)
memory usage: 303.8+ KB


Our data was sourced from the Bank of Ghana Website. It contains about 6478 data points. 

The data is made collected over the span of 26 years.

On 1 July 2007, the country's existing currency, the Cedi (), was converted to the Ghana Cedi (GH), with ten thousand Cedis equaling one Ghana Cedi. The re-denomination was not expected to impact the currency's inherent worth. According to the government, "the value will remain unchanged."



In [19]:

df['Date'] = pd.to_datetime(df['Date'])
df = df.set_index(df['Date'])
df = df.sort_index()

In [22]:
df.isnull().sum()

Date             0
Currency         0
Currency Pair    0
Buying           0
Selling          0
Mid Rate         0
dtype: int64

In [25]:
redenomination = pd.datetime(2007,7,1)

bf_red = df.loc[df['Date'] <= redenomination]
af_red = df.loc[df['Date'] > redenomination]

C:\Users\dzjc9\AppData\Local\Temp\ipykernel_1248\1619372591.py:1: FutureWarning: The pandas.datetime class is deprecated and will be removed from pandas in a future version. Import from datetime module instead.
  redenomination = pd.datetime(2007,7,1)


In [26]:
bf_red

Date   Currency Currency Pair      Buying     Selling  \
Date                                                                     
1996-01-02 1996-01-02  US Dollar        USDGHS  1,434.0900  1,461.6400   
1996-01-03 1996-01-03  US Dollar        USDGHS  1,437.2700  1,464.3600   
1996-01-04 1996-01-04  US Dollar        USDGHS  1,438.6400  1,465.9100   
1996-01-05 1996-01-05  US Dollar        USDGHS  1,442.2700  1,469.0900   
1996-01-08 1996-01-08  US Dollar        USDGHS  1,440.9100  1,467.7300   
...               ...        ...           ...         ...         ...   
2007-06-25 2007-06-25  US Dollar        USDGHS           0           0   
2007-06-26 2007-06-26  US Dollar        USDGHS           0           0   
2007-06-27 2007-06-27  US Dollar        USDGHS  9,213.6400  9,315.8200   
2007-06-28 2007-06-28  US Dollar        USDGHS           0           0   
2007-06-29 2007-06-29  US Dollar        USDGHS           0           0   

              Mid Rate  
Date                    
1996-01-02  1,447.8650  
1996-01-03  1,450.8150  
1996-01-04  1,452.2750  
1996-01-05  1,455.6800  
1996-01-08  1,454.3200  
...                ...  
2007-06-25           0  
2007-06-26           0  
2007-06-27  9,264.7300  
2007-06-28           0  
2007-06-29           0  

[2881 rows x 6 columns]

In [27]:
af_red

Date   Currency Currency Pair  Buying Selling Mid Rate
Date                                                                   
2007-07-03 2007-07-03  US Dollar        USDGHS       0       0        0
2007-07-04 2007-07-04  US Dollar        USDGHS       0       0        0
2007-07-05 2007-07-05  US Dollar        USDGHS       0       0        0
2007-07-06 2007-07-06  US Dollar        USDGHS       0       0        0
2007-07-09 2007-07-09  US Dollar        USDGHS  0.9216  0.9315   0.9266
...               ...        ...           ...     ...     ...      ...
2021-12-30 2021-12-30  US Dollar        USDGHS  5.9931  5.9991   5.9961
2021-12-31 2021-12-31  US Dollar        USDGHS  6.0031  6.0091   6.0061
2022-01-04 2022-01-04  US Dollar        USDGHS  6.0031  6.0091   6.0061
2022-01-05 2022-01-05  US Dollar        USDGHS  6.0029  6.0089   6.0059
2022-01-06 2022-01-06  US Dollar        USDGHS  6.0034  6.0094   6.0064

[3598 rows x 6 columns]